# HG changeset patch
# User =?UTF-8?q?Yoshiki=20V=C3=A1zquez=20Baeza?= <yoshiki89@gmail.com>
# Date 1461443630 25200
#      Sat Apr 23 13:33:50 2016 -0700
# Node ID 754e16ccbd0cc4e2ce096bda11542742dfe25cad
# Parent  b19cab9fe4ad0ffb03b67dd448c9068c234a4b1c
[PATCH] feature: new-mail
This setting executes a command when new mail is received.
---
 curs_main.c | 12 ++++++++++++
 globals.h   |  1 +
 init.h      |  7 +++++++
 3 files changed, 20 insertions(+)

diff -r b19cab9fe4ad -r 754e16ccbd0c PATCHES
--- a/PATCHES	Sat Mar 31 14:15:43 2018 +0200
+++ b/PATCHES	Sat Apr 23 13:33:50 2016 -0700
@@ -1,3 +1,4 @@
+patch-new-mail-neomutt
 patch-notmuch-gentoo
 patch-nntp-vvv
 patch-limit-current-thread-neomutt
diff -r b19cab9fe4ad -r 754e16ccbd0c curs_main.c
--- a/curs_main.c	Sat Mar 31 14:15:43 2018 +0200
+++ b/curs_main.c	Sat Apr 23 13:33:50 2016 -0700
@@ -873,6 +873,12 @@ int mutt_index_menu (void)
 	  mutt_message _("New mail in this mailbox.");
 	  if (option (OPTBEEPNEW))
 	    beep ();
+	  if (NewMailCmd)
+	  {
+	    char cmd[LONG_STRING];
+	    menu_status_line(cmd, sizeof(cmd), menu, NONULL(NewMailCmd));
+	    mutt_system(cmd);
+	  }
 	} else if (check == MUTT_FLAGS)
 	  mutt_message _("Mailbox was externally modified.");
 
@@ -899,6 +905,12 @@ int mutt_index_menu (void)
          menu->redraw |= REDRAW_STATUS;
          if (option (OPTBEEPNEW))
            beep();
+         if (NewMailCmd)
+         {
+           char cmd[LONG_STRING];
+           menu_status_line(cmd, sizeof(cmd), menu, NONULL(NewMailCmd));
+           mutt_system(cmd);
+         }
        }
      }
      else
diff -r b19cab9fe4ad -r 754e16ccbd0c globals.h
--- a/globals.h	Sat Mar 31 14:15:43 2018 +0200
+++ b/globals.h	Sat Apr 23 13:33:50 2016 -0700
@@ -126,6 +126,7 @@ WHERE char *Postponed;
 WHERE char *PostponeEncryptAs;
 WHERE char *Prefix;
 WHERE char *PrintCmd;
+WHERE char *NewMailCmd;
 WHERE char *QueryCmd;
 WHERE char *QueryFormat;
 WHERE char *Realname;
diff -r b19cab9fe4ad -r 754e16ccbd0c init.h
--- a/init.h	Sat Mar 31 14:15:43 2018 +0200
+++ b/init.h	Sat Apr 23 13:33:50 2016 -0700
@@ -322,6 +322,13 @@ struct option_t MuttVars[] = {
   { "beep",		DT_BOOL, R_NONE, OPTBEEP, 1 },
   /*
   ** .pp
+  ** If \fIset\fP, Mutt will call this command after a new message is received.
+  ** See the $$status_format documentation for the values that can be formatted
+  ** into this command.
+  */
+  { "new_mail_command",	DT_PATH, R_NONE, UL &NewMailCmd, UL NULL },
+  /*
+  ** .pp
   ** When this variable is \fIset\fP, mutt will beep when an error occurs.
   */
   { "beep_new",		DT_BOOL, R_NONE, OPTBEEPNEW, 0 },
